functions.php

```diff
@@ -46,4 +50,95 @@ function addAboutMetoDB (PDO $db, string $postdata) : void{
     $query->execute();
 }
 
+/**
+ * takes an array and loops through extracting the first 20 characters of each text input. It then populates them into a dropdown list.
+ *
+ * @param $infos array of db information
+ *
+ * @return string returns a string that can be printed to show a drop down option or options.
+ *
+ */
+function fillEditDropDown (array $infos) :string {
+    $result = '';
+    foreach($infos as $info) {
+        $intro = substr($info['paratext'], 0, 20);
+        $result .= '<option value=' . $info['id'] . '>' . $intro . '</option>';
+    }
+    return $result;
+}
+
+/**
+ *retrieves one result from a db where the id matches the id provided
+ *
+ * @param PDO $db connection to db
+ *
+ * @param $dbEntryId string id and text values to retrieve from db
+ *
+ * @return array returns an array of the information required
+ */
+function getChosenTextToEdit (PDO $db, string $dbEntryId) : array {
+    $query = $db->prepare("SELECT `id`, `paratext` from `about_me` WHERE `id` = '$dbEntryId';");
+    $query->execute();
+    return $query->fetch();
+}
+
+/**
+ * access the value of paratext from any array passed to it
+ *
+ * @param $arr array including a paratext key and value
+ *
+ * @return string contained in the value of the paratext key
+ */
+function retrieveTextFromArray (array $arr) : string {
+    if ($arr['paratext'] == '') {
+        return 'cannot be empty';
```

Comment:
not a change but, this error is coming out when the to be edited paragraph is already empty.